File: python/1457.py
# Definition for a binary tree node.
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
class Solution:
    def pseudoPalindromicPaths (self, root: Optional[TreeNode], mask=0) -> int:
        if root is None:
            return 0
        mask ^= 1 << root.val
        if root.left == root.right:
            return 1 if mask & (mask - 1) == 0 else 0
        return self.pseudoPalindromicPaths(root.left, mask) + self.pseudoPalindromicPaths(root.right, mask)

File: go/1457.go
package main

// type TreeNode struct {
// 	Val   int
// 	Left  *TreeNode
// 	Right *TreeNode
// }

func pseudoPalindromicPaths(root *TreeNode) int {
	return dfs(root, 0)
}

func dfs(root *TreeNode, mask int) int {
	if root == nil {
		return 0
	}
	mask ^= 1 << root.Val
	if root.Left == root.Right {
		if mask&(mask-1) == 0 {
			return 1
		} else {
			return 0
		}
	}
	return dfs(root.Left, mask) + dfs(root.Right, mask)
}
